### PR TITLE
Structured logging

### PR DESCRIPTION
**Pull request by joswarmer**
Replaced logging by using pino,  now we can use multiple log levels.
Also the log output goes both to the console and to the file `server-log.jsonl`.
In the file the format is NDJSON, and has some more fields, which are filtered from the console output top keep it (somewhat) readable.

Also using a Json file for all configuration options. See configuration.md.

@ftomassetti Can you take a look at the text about PGDATABASE in the configuration.md? Not sure it is still relevant.

### CODE REVIEW

File: configuration.md [hunk 1]
```diff
@@ -1,11 +1,68 @@
 # Configuration
 
-Environment variables can be used to configure the project:
+The file server-config.json is used to configure the server:
+It should be in the same folder where the  server is started, if it does not exists, default values will be used"
+
+Below is the server-config.json with all default values
+
+````json
```
Comment (dslmeinte) [inline configuration.md:8]: Shouldn't this be another JSON (sub)format that allows these comments?

File: packages/server/src/server.ts [hunk 2]
```diff
@@ -64,14 +69,29 @@ registerHistoryApi(app, DbConnection.getInstance(), pgp)
 
 const httpServer = http.createServer(app)
 
-const serverPort = parseInt(process.env.NODE_PORT || "3005")
+const serverPort = ServerConfig.getInstance().serverPort()
+
+// Initialize database
```
Comment (dslmeinte) [inline packages/server/src/server.ts:74]: I wouldn't expect these kinds of code changes for a PR that's about (changing) structured logging. Should this be here?

**Review by dslmeinte — APPROVED** [review]

File: configuration.md (hunk 1 above)
Comment (joswarmer) [reply configuration.md:8]: Changed to `json lines` now, was the best available.

File: packages/server/src/server.ts (hunk 2 above)
Comment (joswarmer) [reply packages/server/src/server.ts:74]: The logging needed many more configuration options, therefore I also adjusted the way configuration should be done. So yes, it should be here :-), and yes, it is two changes in the same PR :-(.

Comment (dslmeinte) [reply packages/server/src/server.ts:74]: I meant more the DB initialization code from line 74 onwards, btw.

**Pull request merged**